Makefile.in

```diff
@@ -204,6 +204,9 @@ zlibrc.o: win32/zlib$(SUFFIX)1.rc
 %.lo: $(SRCDIR)/%.c
 	$(CC) $(SFLAGS) -DPIC $(INCLUDES) -c -o $@ $<
 
+$(OBJG): %.o: $(SRCDIR)/%.c
+	$(CC) $(CFLAGS) -DWITH_GZFILEOP $(INCLUDES) -c -o $@ $<
```

Comment:
I think the msan error is due to the fact that we now compile the files with `-DWITH_GZFILEOP`, which exercises a different codepath, so we observe a latent error.

Reply:
This patch fixes the msan error:
```
--- a/gzwrite.c
+++ b/gzwrite.c
@@ -24,10 +24,11 @@ static int gz_init(gz_state *state) {
     state->in = (unsigned char *)malloc(state->want << 1);
     if (state->in == NULL) {
         gz_error(state, Z_MEM_ERROR, "out of memory");
         return -1;
     }
+    memset(state->in, 0, state->want << 1);
 
     /* only need output buffer and deflate state if compressing */
     if (!state->direct) {
         /* allocate output buffer */
         state->out = (unsigned char *)malloc(state->want);
```

Reply:
I have opened pull request #367 with the patch. Thanks for tracking it down for me.

Now I am also wondering if the gz functions should use `zcalloc` instead of `malloc` due to the `UNALIGNED_OK` check.

Reply:
UNALIGNED_OK check only required special handling if vector calls are used with the allocated memory... Default memory alignment is 4 or 8 bytes depending on target, which is enough for normal integer types.